### DISCUSSION TITLE
Return error: -6 Mac M2 Silicon TensorFlow Troubleshooting

### DISCUSSION THREAD

**Discussion by NPabon21** [talmolab/sleap#1430] [Help!] [answered]
I am using SLEAP to track 3 mice with bottom-up processing. I was successful at creating a highly accurate model that works in all frames, but I am having issues predicting many frames at once. I can infer up to 101 frames per try (sometimes only 20 frames), but if I go any higher, I get a “return error: -6” on my Mac Terminal. I have all the latest updates on Python and from what I have read in the terminal, I do not see problems with the size of my RAM. The main issues I see are “Terminating app due to uncaught exception 'NSRangeException’” and “1 leaked semaphore.” Any advice on how to go about this and increase the number of frames we can render in one go would be greatly appreciated.
For more information, I do get the same error code whether it stops me from rendering over 20 frames or over 101 frames. There is also no pattern as to why the number of frames I can predict varies per try. 

Screenshot of Terminal included
<img width="1508" alt="Screenshot 2023-07-27 at 10 54 47 AM" src="https://github.com/talmolab/sleap/assets/140996542/4dd97cf4-f1c8-42b0-bbbc-0fa94c6e2f1b">

**Comment by gitttt-1234** [comment, marked as answer]
The "return error: -6" crash on Mac M2 is a known issue related to TensorFlow's compatibility with Apple Silicon. The `NSRangeException` and leaked semaphore errors point to threading/memory management issues in the TF-Metal backend.

The latest version of SLEAP uses a PyTorch-based backend which has much better Apple Silicon support. We strongly recommend updating to [latest SLEAP v1.6.3](https://docs.sleap.ai/latest/installation/)! 

Closing this discussion, fixed in the latest SLEAP releases!
